In [1]:
%load_ext autoreload
%autoreload 2

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


# Summary

Start compiling humor dataset(s) for RL.

In [2]:
from concurrent.futures import ThreadPoolExecutor
from functools import partial
import numpy as np
import requests
from bs4 import BeautifulSoup
from tqdm.auto import tqdm
import pandas as pd
from datasets import Dataset, load_dataset

from aeon.logging import logger

In [3]:
sess = requests.Session()

In [4]:
def _fetch_standup_transcript(url: str, sess: requests.Session) -> str:
    try:
        r = sess.get(url, timeout=5)
        r.raise_for_status()
    except Exception as e:
        logger.error(f'Failed to fetch url: {url} (error: {e})')
        return ''
    
    soup = BeautifulSoup(r.content, 'lxml')
    return '\n\n'.join(
        para.text 
        for para in soup.select("div.elementor-widget-container > p")
    )

fetch_standup_transcript = partial(_fetch_standup_transcript, sess=sess)

In [5]:
base_url = "https://scrapsfromtheloft.com/stand-up-comedy-scripts/"
url_prefix = 'https://scrapsfromtheloft.com/comedy/'

In [4]:
r = requests.get(base_url)

In [5]:
r

<Response [200]>

In [6]:
soup = BeautifulSoup(r.content, 'lxml')

In [47]:
transcript_urls = list(set(
    [link['href'] for link in soup.find_all('a')
    if link['href'].startswith(url_prefix)
    and link['href'] != url_prefix]
))

In [48]:
len(transcript_urls)

499

In [75]:
with ThreadPoolExecutor(max_workers=20) as exc:
    transcripts = list(tqdm(
        exc.map(fetch_standup_transcript, transcript_urls),
        total=len(transcript_urls)
    ))

  0%|                                                                                                  | 0/499 [00:00<?, ?it/s]ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/stewart-lee-90s-comedian-2006-full-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/stewart-lee-90s-comedian-2006-full-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/ellen-degeneres-the-beginning-2000-full-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/ellen-degeneres-the-beginning-2000-full-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/neal-brennan-women-and-black-dudes-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/neal-brennan-women-and-black-dudes-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/c

ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/fortune-feimster-good-fortune-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/fortune-feimster-good-fortune-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/jim-jefferies-alcoholocaust-2010-full-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/jim-jefferies-alcoholocaust-2010-full-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/dave-chappelle-acceptance-speech-2019-mark-twain-prize/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/dave-chappelle-acceptance-speech-2019-mark-twain-prize/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/eddie-izzard-dress-kill-1999-full-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfr

ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/dave-chappelle-846-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/dave-chappelle-846-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/billy-connolly-high-horse-tour-live-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/billy-connolly-high-horse-tour-live-transcript/)
 11%|█████████▋                                                                               | 54/499 [00:03<00:32, 13.72it/s]ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/mo-amer-mohammed-in-texas-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/mo-amer-mohammed-in-texas-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/chelsea-handler-the-feeling-transcript/ (error: 503 

ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/ralphie-may-filthy-animal-tour-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/ralphie-may-filthy-animal-tour-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/john-mulaney-snl-monologue-2018-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/john-mulaney-snl-monologue-2018-transcript/)
 17%|███████████████▌                                                                         | 87/499 [00:04<00:13, 31.21it/s]ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/george-carlin-dumb-americans-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/george-carlin-dumb-americans-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/gary-gulman-born-on-3rd-base-t

ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/sammy-obeid-how-to-save-gaza-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/sammy-obeid-how-to-save-gaza-transcript/)
 24%|████████████████████▊                                                                   | 118/499 [00:04<00:08, 43.59it/s]ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/hannah-berner-we-ride-at-dawn-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/hannah-berner-we-ride-at-dawn-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/eddie-izzard-glorious-1997-full-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/eddie-izzard-glorious-1997-full-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/sebastian-maniscalco-whats-wrong

ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/tom-segura-disgraceful-2018-full-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/tom-segura-disgraceful-2018-full-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/kevin-hart-zero-fks-given-2020-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/kevin-hart-zero-fks-given-2020-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/hannah-gadsby-nanette-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/hannah-gadsby-nanette-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/bert-kreischer-fighting-a-bear-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/bert-kreischer-fighting-a-bear-tra

ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/drew-michael-red-blue-green-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/drew-michael-red-blue-green-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/jim-gaffigan-quality-time-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/jim-gaffigan-quality-time-transcript/)
 36%|███████████████████████████████▋                                                        | 180/499 [00:06<00:06, 51.82it/s]ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/hasan-minhaj-homecoming-king-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/hasan-minhaj-homecoming-king-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/iliza-shlesinger-unveiled-transcript/ (error: 50

ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/ali-wong-baby-cobra-2016-full-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/ali-wong-baby-cobra-2016-full-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/richard-pryors-monologue-saturday-night-live-1975/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/richard-pryors-monologue-saturday-night-live-1975/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/cristela-alonzo-middle-classy-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/cristela-alonzo-middle-classy-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/jim-jefferies-intolerant-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/jim-jefferies-

ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/gabriel-iglesias-hot-and-fluffy-2007-full-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/gabriel-iglesias-hot-and-fluffy-2007-full-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/kevin-james-never-dont-give-up-full-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/kevin-james-never-dont-give-up-full-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/russell-brand-messiah-complex-2013-full-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/russell-brand-messiah-complex-2013-full-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/kevin-bridges-the-story-continues-transcript/ (error: 503 Server Error: Service Unavailable for url: http

ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/phil-wang-philly-philly-wang-wang-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/phil-wang-philly-philly-wang-wang-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/roy-wood-jr-no-one-loves-you-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/roy-wood-jr-no-one-loves-you-transcript/)
 53%|██████████████████████████████████████████████▉                                         | 266/499 [00:08<00:07, 32.36it/s]ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/jim-gaffigan-dark-pale-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/jim-gaffigan-dark-pale-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/cristela-alonzo-lower-classy-2017-full-tra

ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/leanne-morgan-im-every-woman-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/leanne-morgan-im-every-woman-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/rory-scovel-tries-stand-up-for-the-first-time-a-netflix-special/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/rory-scovel-tries-stand-up-for-the-first-time-a-netflix-special/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/taylor-tomlinson-have-it-all-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/taylor-tomlinson-have-it-all-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/iliza-shlesinger-elder-millennial-2018-full-transcript/ (error: 503 Server Error: Service Unavailable for url: https://

ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/colin-quinn-the-new-york-story-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/colin-quinn-the-new-york-story-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/tim-dillon-this-is-your-country-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/tim-dillon-this-is-your-country-transcript/)
 66%|██████████████████████████████████████████████████████████▎                             | 331/499 [00:10<00:04, 33.81it/s]ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/bo-burnham-what-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/bo-burnham-what-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/fred-armisen-standup-for-drummers-transcript/ (error: 50

ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/george-carlin-you-are-all-diseased-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/george-carlin-you-are-all-diseased-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/michelle-wolf-the-well-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/michelle-wolf-the-well-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/kevin-bridges-whole-different-story-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/kevin-bridges-whole-different-story-transcript/)
 72%|███████████████████████████████████████████████████████████████▎                        | 359/499 [00:11<00:04, 30.22it/s]ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/patton-oswalt-we-all-screa

ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/dave-chappelle-hbo-half-hour-1998-traduzione-italiana/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/dave-chappelle-hbo-half-hour-1998-traduzione-italiana/)
 78%|████████████████████████████████████████████████████████████████████▌                   | 389/499 [00:12<00:02, 38.88it/s]ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/w-kamau-bell-private-school-negro-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/w-kamau-bell-private-school-negro-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/amy-schumer-growing-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/amy-schumer-growing-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/eric-andre-legalize-

ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/mike-epps-only-one-mike-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/mike-epps-only-one-mike-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/enissa-amani-ehrenwort-2018-full-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/enissa-amani-ehrenwort-2018-full-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/daniel-sloss-x-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/daniel-sloss-x-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/david-spade-dandelion-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/david-spade-dandelion-transcript/)
ERROR:aeon.logging:Failed to fetch u

ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/kevin-james-irregardless-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/kevin-james-irregardless-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/nate-bargatze-nashville-christmas-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/nate-bargatze-nashville-christmas-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/t-j-miller-no-real-reason-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/t-j-miller-no-real-reason-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/louis-c-k-sorry-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/louis-c-k-sorry-transcript/)
ERROR:aeon.logging:Fai

ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/theo-von-no-offense-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/theo-von-no-offense-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/jimmy-carr-funny-business-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/jimmy-carr-funny-business-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/sam-morril-youve-changed-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/sam-morril-youve-changed-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scrapsfromtheloft.com/comedy/mae-martin-sap-transcript/ (error: 503 Server Error: Service Unavailable for url: https://scrapsfromtheloft.com/comedy/mae-martin-sap-transcript/)
ERROR:aeon.logging:Failed to fetch url: https://scra

In [76]:
sum(bool(row) for row in transcripts)

7

In [69]:
pd.DataFrame({"url": transcript_urls, "transcript": transcripts}).iloc[10:50]

url  \
10  https://scrapsfromtheloft.com/comedy/kevin-bri...   
11  https://scrapsfromtheloft.com/comedy/ali-wong-...   
12  https://scrapsfromtheloft.com/comedy/stewart-l...   
13  https://scrapsfromtheloft.com/comedy/adam-devi...   
14  https://scrapsfromtheloft.com/comedy/trevor-no...   
15  https://scrapsfromtheloft.com/comedy/dave-atte...   
16  https://scrapsfromtheloft.com/comedy/nate-barg...   
17  https://scrapsfromtheloft.com/comedy/neal-bren...   
18  https://scrapsfromtheloft.com/comedy/ellen-deg...   
19  https://scrapsfromtheloft.com/comedy/hannah-ei...   
20  https://scrapsfromtheloft.com/comedy/david-cro...   
21  https://scrapsfromtheloft.com/comedy/stewart-l...   
22  https://scrapsfromtheloft.com/comedy/sebastian...   
23  https://scrapsfromtheloft.com/comedy/george-lo...   
24  https://scrapsfromtheloft.com/comedy/deon-cole...   
25  https://scrapsfromtheloft.com/comedy/larry-the...   
26  https://scrapsfromtheloft.com/comedy/fortune-f...   
27  https://scrapsfromtheloft.com/comedy/dave-chap...   
28  https://scrapsfromtheloft.com/comedy/chris-roc...   
29  https://scrapsfromtheloft.com/comedy/jim-jeffe...   
30  https://scrapsfromtheloft.com/comedy/eddie-gri...   
31  https://scrapsfromtheloft.com/comedy/dave-chap...   
32  https://scrapsfromtheloft.com/comedy/mike-epps...   
33  https://scrapsfromtheloft.com/comedy/ramy-yous...   
34  https://scrapsfromtheloft.com/comedy/tom-papa-...   
35  https://scrapsfromtheloft.com/comedy/dick-greg...   
36  https://scrapsfromtheloft.com/comedy/langston-...   
37  https://scrapsfromtheloft.com/comedy/eddie-izz...   
38  https://scrapsfromtheloft.com/comedy/jack-whit...   
39  https://scrapsfromtheloft.com/comedy/george-ca...   
40  https://scrapsfromtheloft.com/comedy/jerry-sei...   
41  https://scrapsfromtheloft.com/comedy/oh-hello-...   
42  https://scrapsfromtheloft.com/comedy/beth-stel...   
43  https://scrapsfromtheloft.com/comedy/lee-mack-...   
44  https://scrapsfromtheloft.com/comedy/iliza-shl...   
45  https://scrapsfromtheloft.com/comedy/jo-koy-li...   
46  https://scrapsfromtheloft.com/comedy/jeff-dunh...   
47  https://scrapsfromtheloft.com/comedy/jimmy-car...   
48  https://scrapsfromtheloft.com/comedy/trevor-no...   
49  https://scrapsfromtheloft.com/comedy/ricky-ger...   

                                           transcript  
10  ♪ I’m at the crossroads ♪\n♪ Getting drowned i...  
11  [“Get me Bodied (Extended Mix)” playing]\n\n[a...  
12  Recorded on 10 March 2006 at Chapter Arts, Can...  
13  [rock music playing]\n\n[indistinct chatter]\n...  
14  Filmed at The Lincoln Theatre in Washington, D...  
15  [Cheers and applause] – you guys ready to meet...  
16  Nate Bargatze: Hello World (2023)\nGenre: Come...  
17  [Hip-hop music plays] [cheers and applause] ma...  
18  Ladies and gentlemen, please, welcome Ellen De...  
19  (vinyl record popping)\n\n(“J’ai Du L’Oublier”...  
20  David Cross: Worst Daddy in the World\n\nFilme...  
21  This programme contains very strong language a...  
22  Sebastian Maniscalco returns to The Tonight Sh...  
23  San Francisco, it’s my honor to introduce to y...  
24  [indistinct chattering]\n\n[woman] Oh, this wa...  
25  [Announcer] Ladies and gentlemen, Larry, The C...  
26  [upbeat music plays]\n\n[audience cheering]\n\...  
27  An outstanding lineup of entertainers gathered...  
28  Ladies and gentlemen… live from the world-famo...  
29  [Horn honks] [Indistinct conversations] [scatt...  
30  [♪♪♪] [notification beep] [♪♪♪]\n\n[cheers and...  
31  Original air date: November 07, 2020\n\n \n\n ...  
32  [Cheers and applause]\n\nHoo! Hoo! Hoo!\n\nHoo...  
33  Ramy Youssef: More Feelings (2024)\n\nReleased...  
34  aired December 2016\n\n[plodding music]\n\n♪\n...  
35  George O’Hare recounts his friendship with Dic...  
36                                                     
37                                                     
38  Jack Whitehall: Settle Down (2024)\n\nIn “Jack...  
39                              

Hitting rate limits. Could slow down dramatically + add backoff, orrr just revert to the existing huggingface dataset. Was going to do this myself for fun but if it's going to be a pain maybe not worth it, huggingface version has >80% of them.

## Huggingface dataset

In [5]:
ds = load_dataset("zachgitt/comedy-transcripts")

In [6]:
df = ds['train'].to_pandas()\
    .rename(columns=lambda x: x.replace('-', '_'))

In [11]:
df.transcript.sample().values[0]

'Air date: August 14, 2015\n Recorded at The Lincoln Theater in Washington D.C. [rock music plays] [crowd cheering] Thank you. Thank you so much, everybody. Thank you for clapping. I appreciate that. Thank you. Thanks. Thanks a lot. Thank you. That’s cool. All right. This is a good crowd. This is… All right. Thank you. That guy has a high voice. – [laughter] – [laughs] All right, I got a lot of jokes, so I’m just gonna start telling them. Here we go. I went to a baseball game last summer in a stadium and they had a huge TV in the stadium. A Jumbotron. And this guy proposed to his girlfriend using the giant TV. He put her name up there, said, “Will you marry me?” She said yes. The crowd went wild. They found the couple in the audience. I was sitting there thinking, “God, that’s so romantic. That’s so cool.” And then I remembered thinking, you know, you could also use a screen like that if you’re having trouble breaking up with somebody. Be like, “Hey, I’m gonna grab a hot dog. But you s

In [52]:
df.loc[[257], ['transcript']]

transcript
257  Air date: August 14, 2015\n Recorded at The Li...

In [39]:
# Approximate transcript token counts
df.transcript.str.len().div(4).describe()

count      419.000000
mean     10993.677804
std       4452.077710
min        646.750000
25%       9332.750000
50%      11620.250000
75%      13910.375000
max      23058.500000
Name: transcript, dtype: float64

In [40]:
# Approximate transcript token counts
df.transcript.str.len().div(4).sum()

np.float64(4606351.0)

In [7]:
# Super rough napkin math, rounded everything
tokens_per_transcript = 10_000
n_transcripts = 400
n_100_token_calls = tokens_per_transcript * n_transcripts / 100
kb_per_100_token_call = np.array([10, 40, 100, 300])

kb_per_gb = 1_000_000
total_gb = n_100_token_calls * kb_per_100_token_call / kb_per_gb
total_gb

array([ 0.4,  1.6,  4. , 12. ])

Looking for possible chunking strategies (lot of models with limited output lengths that will force us to label each transcript in several goes).

In [79]:
for term in ('[laugh]', '[laughs]', '[laughter]', '[pause]', '\n', '\n\n'):
    print(
        repr(term),
        df.transcript.apply(lambda x: x.count(term)).value_counts(normalize=True).head(),
        end='\n\n'
    )

'[laugh]' transcript
0    0.997613
1    0.002387
Name: proportion, dtype: float64

'[laughs]' transcript
0    0.758950
1    0.073986
2    0.023866
6    0.023866
3    0.021480
Name: proportion, dtype: float64

'[laughter]' transcript
0    0.825776
1    0.035800
2    0.016706
7    0.009547
3    0.009547
Name: proportion, dtype: float64

'[pause]' transcript
0    0.995227
4    0.002387
1    0.002387
Name: proportion, dtype: float64

'\n' transcript
0    0.431981
1    0.095465
2    0.059666
4    0.038186
3    0.031026
Name: proportion, dtype: float64

'\n\n' transcript
0    1.0
Name: proportion, dtype: float64



In [87]:
len(df.transcript.values[0].split('. '))

889

## Labeling

Scratch code working out llm labeling stuff.

**Idea for later:** logprobs might actually be a really promising data source here to create more variants cheaply. Swap out some low probability word for a high probability one, joke ruined.

In [94]:
from aeon.labeling import LLMLabeler
from aeon.prompt import list_prompts, Prompt, Prompts
from aeon.logging import logger

In [104]:
# p = Prompt(Prompts.EXTRACT_JOKES, model="gpt-5-nano", top_logprobs=None)

In [221]:
llm = LLMLabeler(
    Prompts.EXTRACT_JOKES,
)

In [222]:
res = llm.label(
    df.loc[[257], ['transcript']],
    # df.tail(15).assign(transcript=lambda x: x.transcript.str[:1_000]),
    model='gpt-5-mini', top_logprobs=None
)

Unstaged/uncommitted changes in git repository. You might want to commit them before running.
Labels will be saved in /Users/hmamin/aeon/aeon/data/labels/extract_jokes/2025.11.16_21.49.15-73584418f4d559083d41bd804cbc5f29489bc75c


Labeling rows:   0%|          | 0/1 [00:00<?, ?it/s]

[TIMER] BLOCK executed in 243.274 s.
Removing intermediate results dir since job completed without interruption.


In [223]:
dfl = res['df']

In [224]:
dfl.response_content.str['items'].str.len()

0    132
Name: response_content, dtype: int64

In [225]:
row = dfl.sample()
print(row.index)
row.response_content.iloc[-1]

Index([0], dtype='int64')


{'items': [{'joke': 'I went to a baseball game last summer in a stadium and they had a huge TV in the stadium. A Jumbotron. And this guy proposed to his girlfriend using the giant TV. He put her name up there, said, “Will you marry me?” She said yes. The crowd went wild. I was sitting there thinking, “God, that’s so romantic. That’s so cool.” And then I remembered thinking, you know, you could also use a screen like that if you’re having trouble breaking up with somebody. Be like, “Hey, I’m gonna grab a hot dog. But you should definitely look at that screen.” That’s a smooth way out of it.',
   'prompt': 'What would you do with a Jumbotron at a stadium?',
   'subtext': 'Large public screens can be used for dramatic public gestures, including proposals or breakups.',
   'unfunny_variant': 'I saw someone propose on the Jumbotron and thought you could also announce a breakup on it by pointing people to the screen.'},
  {'joke': 'I think my favorite sign that I’ve seen is “Bridge May Be Ic

In [226]:
# Checking how many chars of transcript it kept. Looks like a very low %, 17/45, but
# in practice it stopped about 26k chars into the transcript so that looks like a reasonable % to me.
sum(len(line['joke']) for line in row.response_content.iloc[-1]['items'])

36943

In [227]:
len(row.api_kwargs.values[0]['messages'][-1]['content'])

45086

In [228]:
row.response_raw.values[0]['choices'][0]['finish_reason']

'stop'

In [230]:
# View all extracted jokes
# print('\n\n'.join(line['joke'] for line in row.response_content.iloc[-1]['items']))

In [234]:
term = 'Yep, yep, yep!'
x, y = row.api_kwargs.values[0]['messages'][-1]['content'].split(term)

In [235]:
len(x)

44890

In [236]:
len(y)

182

In [186]:
dfl.response_raw.values[0]['id']

'chatcmpl-CcShI8IgCq5bWUxmCSNEt2BkhrNh4'

In [239]:
df.shape[0] * dfl.response_raw.values[0]['usage']['prompt_tokens']

5399653

In [238]:
df.shape[0] * dfl.response_raw.values[0]['usage']['completion_tokens']

7254147

In [240]:
dfl.response_raw.values[0]['usage']['completion_tokens']

17313

In [190]:
df.transcript.str.len().quantile([0, .1, .2, .5, .8, .9, .95, .99, 1])

0.00     2587.00
0.10    12012.20
0.20    33403.80
0.50    46481.00
0.80    57304.40
0.90    62747.20
0.95    67580.50
0.99    84103.42
1.00    92234.00
Name: transcript, dtype: float64

In [196]:
# json.dumps(dfl.api_kwargs.iloc[0]['messages'])

run | jokes | char_count_jokes | approx char_count_transcript | output_tokens
---|---|---|---|---
original | 60 | 17.1k | 25.8k | 7.6k
ask nicely | 132 | 36.9k | 44.9k | 17.3k

## Scratch

Older code working out early labeling functionality.

In [100]:
from aeon.secrets import SecretManager
from openai import OpenAI
from pydantic import BaseModel, Field, create_model
import re

In [101]:
class Response(BaseModel):
    
    joke: str = Field(
        ...,
        description="One or more sentences containing a joke from the "
                    "input text."
    )
        
    subtext: str = Field(
        ...,
        description="The often banal observation underlying the joke. "
                    "This should not be funny."
    )
        
    joke_variant: str = Field(
        ...,
        description="Your attempt to modify the original joke while "
                    "maintaining the same subtext."
    )

In [102]:
Batch = create_model('Batch', items=list[Response])

In [103]:
instructions = """
You are a detail-oriented research assistant with a shrewd understanding of humor, human behavior, and writing.
Your job is to extract jokes from the user's message.

  <instructions>You are one step in a data pipeline for a humor-related research project. Your task is to extract all jokes from the input passage and return valid JSON where each object contains "joke" and "subtext" fields. You can think of the subtext as the often banal point that the joke is making, which the comedian has ultimately massaged or restructured such that the resulting joke subverts the audience's expectations in a fun way. The subtext should not be funny, e.g. it might plainly state "airplane food is bad". Not every sentence in the input must belong to a joke, but most of them probably will because I am showing you standup transcripts.
      
  <example_input>
  [comedian: Louis CK]
  I could never wear white pants because I’ll get my period, first of all. I know that. [Laughter] Or diarrhea, more likely. [Laughter] Which is — That’s really my period. Diarrhea. About once a month, I’m like, “Oh, fuck, here we go.” [Laughter] “Better just get home. And don’t make any big decisions today.” [Laughter] It’s true. Don’t make — You know, if you have diarrhea, don’t, like, negotiate. It’s a bad bargaining position. If I have diarrhea, you stand between me and the toilet, I’ll sell you my house for 10 cents.
  </example_input>
  <example_output>
  [
      {
          "joke": "I could never wear white pants because I’ll get my period, first of all. I know that. Or diarrhea, more likely.",
          "subtext": "Wearing white pants makes any bodily mishap immediately obvious, so it’s risky."
      },
      {
          "joke": "Which is — That’s really my period. Diarrhea. About once a month, I’m like, “Oh, fuck, here we go.”",
          "subtext": "Monthly digestive mishaps can feel as routine and unavoidable as menstruation."
      },
      {
          "joke": "“Better just get home. And don’t make any big decisions today.” It’s true. Don’t make — You know, if you have diarrhea, don’t, like, negotiate. It’s a bad bargaining position. If I have diarrhea, you stand between me and the toilet, I’ll sell you my house for 10 cents.",
          "subtext": "Being desperate for a bathroom gives you zero leverage in any negotiation."
      }
  ]
  </example_output>
  </instructions>
"""

In [104]:
# number of sentences in one mulaney transcript
len(df.transcript.iloc[418].split('.'))

863

In [105]:
unlabeled = '.'.join(df.transcript.iloc[418].split('.')[:25]) + '.'
user_message = f"<example_to_label>{unlabeled}</example_to_label>"

In [106]:
secrets = SecretManager().get_secrets()

In [107]:
openai_client = OpenAI(api_key=secrets['OPENAI_API_KEY'])

In [108]:
tmp = openai_client.chat.completions.parse(
    model="gpt-4.1-nano",
    messages=[
        {"role": "developer", "content": instructions},
        {"role": "user", "content": user_message}
    ],
    response_format=Batch,
    temperature=0.3,
    logprobs=True,
    top_logprobs=20,
)

In [109]:
type(tmp)

openai.types.chat.parsed_chat_completion.ParsedChatCompletion[Batch]

In [125]:
tmp.model_dump(mode='json')['choices'][0]['message']['content']

'{"items":[{"joke":"The past couple years, I’ve done a lot of work on myself. And I’ve realized that I’ll be fine as long as I get constant attention.","subtext":"The speaker values attention highly and believes it is essential for his well-being.","joke_variant":"I\'ve learned that I need constant attention to be okay."},{"joke":"When I was three years old, they pulled me aside and they told me that I was adopted. And that my real mother had been murdered… by Miss America.","subtext":"The speaker was told a fabricated, sensational story about his adoption and mother\'s death.","joke_variant":"As a child, I was told I was adopted and that my mother was murdered by Miss America."},{"joke":"They said, \'If you ask our mom about it, she’ll get really upset. So don’t ask her if it’s true unless you want to upset her.\' And they said, \'If you ask our dad about it, he’ll say that we’re lying.\' But he’s lying.","subtext":"The children were advised to avoid asking their parents about the fab

In [62]:
tmp = {
    "index": 0,
    "success": True,
    "error": "",
    "response": res.model_dump_json(),
    "api_kwargs": {},
}

In [63]:
with open('/tmp/0.json', "w") as f:
    json.dump(tmp, f)

Observation: joke variant actually maybe works better than subtext! Using kinda dumb models maybe works pretty well, you don't need to tell it to make the joke unfunny, it does it naturally!

In [40]:
json.loads(res.choices[0].message.content)['items']

[{'joke': 'The past couple years, I’ve done a lot of work on myself. And I’ve realized that I’ll be fine as long as I get constant attention.',
  'subtext': 'People often seek attention to feel okay about themselves.',
  'joke_variant': "I've learned that I need constant attention to feel okay."},
 {'joke': 'When I was three years old, they pulled me aside and they told me that I was adopted. And that my real mother had been murdered… by Miss America.',
  'subtext': 'Children can misunderstand complex family situations in humorous ways.',
  'joke_variant': "As a child, I misunderstood my family's explanations about my adoption and my mother's death."},
 {'joke': 'They thought of every angle.',
  'subtext': 'People often prepare for all possible questions or reactions.',
  'joke_variant': 'They considered every possible way I might react or ask questions.'},
 {'joke': 'And to compound the stress that I was under, when I was three years old, I thought that Miss America was the Statue of 

In [50]:
json.loads(res.model_dump_json()).keys()

dict_keys(['id', 'choices', 'created', 'model', 'object', 'service_tier', 'system_fingerprint', 'usage'])

In [63]:
res.usage

CompletionUsage(completion_tokens=508, prompt_tokens=1165, total_tokens=1673, completion_tokens_details=CompletionTokensDetails(accepted_prediction_tokens=0, audio_tokens=0, reasoning_tokens=0, rejected_prediction_tokens=0), prompt_tokens_details=PromptTokensDetails(audio_tokens=0, cached_tokens=0))

In [175]:
# [row for row in res.choices[0].logprobs.content]

In [10]:
df[['transcript']].head().to_dict(orient='records')

[{'transcript': 'Full transcript of It’s Bad for Ya, final HBO stand-up comedy special by stand-up comedian George Carlin. It was televised live on March 1, 2008 on HBO. Filmed in the Wells Fargo Center for the Arts in Santa Rosa, California Thank you. Thank you. Thank you. I’d like to begin… I’d like to… Thank you. Thank you. I’d like to begin by saying fuck Lance Armstrong. Fuck him and his balls and his bicycles and his steroids and his yellow shirts and the dumb, empty expression on his face. I’m tired of that asshole. And while you’re at it fuck Tiger Woods, too. There’s another jack-off I can do without. I’m tired of being told who to admire in this country. Aren’t you sick of being told who your heroes ought to be? You know? Being told who you ought to be looking up to. I’ll choose my own heroes, thank you very much. And fuck Dr. Phil, too. Dr. Phil said I should express my emotions, so that’s what I’m doing. Now, since the last time I rolled through these parts, and I do roll t